gusto/transport_equation.py

```diff
@@ -366,18 +367,18 @@ def advection_term(self, q):
             both = lambda u: 2*avg(u)
 
             L = (
-                inner(q, curl(cross(self.ubar, self.test)))*dx
-                - inner(both(self.Upwind*q),
+                inner(self.P(q), curl(cross(self.ubar, self.test)))*dx
+                - inner(both(self.Upwind*self.P(q)),
```

Comment:
If `q` should always be projected, better to apply the projection when you define `q`.

Reply:
It's not projected in the case of integration by parts twice, because I have not tested this case. In theory the projection still should be applied to the all ```q```s in this case, so it's probably fine to leave all ```q```s as they are and do something like
```q = self.P(q)```
at the beginning of ```advection_term```.